app/models/foreman_docker/docker.rb

```diff
@@ -69,13 +69,18 @@ def provider_friendly_name
     end
 
     def create_container(args = {})
-      options = vm_instance_defaults.merge(args)
+      options = args.merge(vm_instance_defaults)
       logger.debug("Creating container with the following options: #{options.inspect}")
-      ::Docker::Container.create(options)
-    rescue Excon::Errors::Error, ::Docker::Error::DockerError => e
-      logger.debug "Fog error: #{e.message}\n " + e.backtrace.join("\n ")
-      errors.add(:base, _("Error creating container. Check the Foreman logs: %s") % e.message.to_s)
-      false
+      docker_command do
+        ::Docker::Container.create(options)
+      end
+    end
+
+    def create_image(args = {})
+      logger.debug("Creating docker image with the following options: #{args.inspect}")
+      docker_command do
+        ::Docker::Image.create(args)
+      end
     end
 
     def vm_instance_defaults
```

Comment:
Unrelated to the PR, but while you're at solving the issue a few lines above, could you change this 'cmd' to 'Cmd'? Otherwise it's not setting the default, just adding a new key to the hash. Or just remove it as images provide a default entrypoint already, do as you wish...